Fix vecdot implementation edge cases

See #399 and the failure in array-api-tests:

```
E       ZeroDivisionError: division by zero
E       Falsifying example: test_vecdot(
E           x1=cubed.Array<array-212, shape=(0,), dtype=uint8, chunks=((0,),)>,
E           x2=cubed.Array<array-213, shape=(0,), dtype=uint8, chunks=((0,),)>,
E           data=data(...),
E       )
E       Draw 1: {}
```

Our implementation should also be brought up to date with the [one in array-api-compat](https://github.com/data-apis/array-api-compat/blob/40603a991d362aa02183aac4c8ad342e034112e0/array_api_compat/common/_aliases.py#L491-L508). (I tried doing this, but still got the test failure.)

A similar [failure](https://github.com/cubed-dev/cubed/actions/runs/8074828830/job/22060739025) is now occurring in `tensordot`:

```
FAILED array_api_tests/test_linalg.py::test_tensordot - ZeroDivisionError: division by zero
Falsifying example: test_tensordot(
    x1=cubed.Array<array-1074, shape=(0, 0), dtype=uint8, chunks=((0,), (0,))>,
    x2=cubed.Array<array-1075, shape=(0, 0), dtype=uint8, chunks=((0,), (0,))>,
    kw={},
)

You can reproduce this example by temporarily adding @reproduce_failure('6.98.13', b'AXicY2BgYARCCAECAAA7AAU=') as a decorator on your test case
```

--- a/cubed/tests/test_array_api.py
+++ b/cubed/tests/test_array_api.py
@@ -194,6 +194,7 @@ def test_negative(spec, executor):
         slice(None),
         slice(10),
         slice(3, None),
+        slice(3, 3),
         slice(3, 10),
         (slice(10), None),  # add a new dimension
     ],
@@ -208,11 +209,12 @@ def test_index_1d(spec, ind):
     [
         (2, 3),
         (None, 2, 3),  # add a new dimension
+        (slice(None), slice(2, 2)),
         (slice(None), slice(2, 4)),
         (slice(3), slice(2, None)),
         (slice(1, None), slice(4)),
+        (slice(1, 1), slice(None)),
         (slice(1, 3), slice(None)),
-        (slice(None), slice(2, 4)),
         (None, slice(None), slice(2, 4)),  # add a new dimension
         (slice(None), None, slice(2, 4)),  # add a new dimension
         (slice(None), slice(2, 4), None),  # add a new dimension

--- a/cubed/tests/test_utils.py
+++ b/cubed/tests/test_utils.py
@@ -44,6 +44,7 @@ def test_block_id_to_offset():
 
 def test_to_chunksize():
     assert to_chunksize(((3, 3, 3, 1),)) == (3,)
+    assert to_chunksize(((0,),)) == (1,)  # Zarr doesn't support zero-length chunks
     with pytest.raises(ValueError):
         to_chunksize(((3, 2, 3, 3, 1),))
 

--- a/cubed/utils.py
+++ b/cubed/utils.py
@@ -122,7 +122,7 @@ def to_chunksize(chunkset: T_RectangularChunks) -> T_RegularChunks:
     if not _check_regular_chunks(chunkset):
         raise ValueError(f"Array must have regular chunks, but found chunks={chunkset}")
 
-    return tuple(c[0] for c in chunkset)
+    return tuple(max(c[0], 1) for c in chunkset)
 
 
 @dataclass

